### ISSUE TITLE
Volume umounted after kubelet restart

### ISSUE THREAD

**Issue by valentin2105** [kubernetes/kubernetes#53246]
**What happened**:

When upgrading k8s on the 1.7.x branch, (1.7.4 to 1.7.6 exactly), every Kubernetes services can be restarted without problems except for Kubelet that cause volumes get umounted in pods.

I use GlusterFS volumes and when I restart Kubelet the volume is unmounted and it case a crash of the applications.

```
        volumeMounts:
        - name: db-folder
          mountPath: /var/lib/mysql
      volumes:
        - name: db-folder
          glusterfs:
            endpoints: glusterfs-cluster
            path: website-db
            readOnly: false
```

```
root@website-db-3276445667-gl5nv:/# df -h
df: /var/lib/mysql: Transport endpoint is not connected
root@nwebsite-db-3276445667-gl5nv:/# cd /var/lib/mysql
bash: cd: /var/lib/mysql: Transport endpoint is not connected
```
**What you expected to happen**:
Kubelet need to be able to restart without this issue. 

**How to reproduce it (as minimally and precisely as possible)**:
- Use external volumes (GlusterFS but other can be impacted? ),

- Go on the worker where your pod run, 

- Restart Kubelet,

- Check if your application is still up. 

Kubernetes version (use `kubectl version`):

1.7.6

Cloud provider or hardware configuration**:

Debian 9, multi nodes, IPv4/6 with Calico, on premise. 

/kind bug

**Comment by gnufied** [comment]
This could very well be related to https://github.com/kubernetes/kubernetes/pull/49640 . @valentin2105 will you be able to pull in linked PR and see if that fixes your problem?

**Comment by dchen1107** [comment]
cc/ @jsafrane @jingxu97

**Comment by valentin2105** [comment]
@gnufied I'm not really able to rebuild kubelet from the merged PR (never do that) and I think is not easy as `go build`.
Any idea on which release it would be merged ? in 1.7.x ?
But the PR discussion seem exactly what happened to me.

**Comment by jsafrane** [comment]
@valentin2105, how do you (re)start kubelet? Do you run it in a system services? #49640 will help you if so. It's available in 1.8 branch. I am afraid that draining the node before upgrading it is necessary in 1.7.x if you use Gluster.

**Comment by valentin2105** [comment]
Hi @jsafrane , Yep, I use systemd units. 
Do 1.8.1 will got the PR on it ?

Thanks, I will deal with that in 1.7 and wait to upgrade in 1.8 ;)
